Add inline call to profiling executor pipeline

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43743 Update aliasing in tensorexpr fuser
* #43742 use types in the IR instead of vmap
* **#43636 Add inline call to profiling executor pipeline**
* #43635 Use prim::TensorExprGroup interned symbol
* #43634 Update requires grad property

We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

Commit:
Add passes to profiling executor pipeline

[ghstack-poisoned]

Commit:
Update on "Add passes to profiling executor pipeline"




[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:
```diff
@@ -313,11 +315,13 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
 
 void ProfilingGraphExecutorImpl::runProfilingInsensitiveOptimizations(
     std::shared_ptr<Graph>& graph) {
+  Inline(*graph);
```
Nit: might be worth adding a comment why we need it. I assume it's because we're now planning to generate function calls for fallbacks (if I didn't know that I'd assume this pass was useless)?

eellison replied:
We weren't running inlining in the forward graph of differentiable subgraphs at all

eellison replied:
Previously we would have 
```
%514 : Function = prim::Constant[name="AD_mm_backward_self"]()
grad_self.54 : Tensor = prim::CallFunction(%514, %868, %17) # <string>:204:28
```

ZolotukhinM commented on torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:
```diff
@@ -313,11 +315,13 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
     runNoGradOptimizations(copy);
   }
   EliminateDeadCode(copy);
+  ClearProfilingInformation(copy);
```
Why do we now need this?
Also, please add a `GRAPH_DUMP`statement before it.

eellison replied:
Yea, I think this locally before the rebase, it's not needed since the tensorexpr fuser will remove the profiles.

ZolotukhinM commented:
I'm not sure I fully understand motivation of adding these passes (even after re-reading the description in the commit message). Could you please elaborate a bit more on why it wasn't needed before and why we need it now?

Commit:
Update on "Add passes to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs, and we weren't getting rid of all profiles as part of optimization.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
✌️

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]

Commit:
Update on "Add inline call to profiling executor pipeline"


We weren't running inlining in the forward graph of differentiable subgraphs.

Differential Revision: [D23358804](https://our.internmc.facebook.com/intern/diff/D23358804)

[ghstack-poisoned]
